openssl-dynamic/src/main/c/ssl.c

```diff
@@ -2060,13 +2146,33 @@ TCN_IMPLEMENT_CALL(jlong, SSL, parseX509Chain)(TCN_STDARGS, jlong x509ChainBio)
 
     UNREFERENCED(o);
 
+#ifdef OPENSSL_IS_BORINGSSL
+    while (PEM_read_bio(cert_bio, &name, &header, &data, &data_len)) {
+
+        OPENSSL_free(name);
+        name = NULL;
+
+        OPENSSL_free(header);
+        header = NULL;
+
+        buffer = CRYPTO_BUFFER_new(data, data_len, NULL);
+        OPENSSL_free(data);
+        data = NULL;
+
+        if (buffer == NULL || sk_CRYPTO_BUFFER_push(chain, buffer) <= 0) {
+#else
     chain = sk_X509_new_null();
     while ((cert = PEM_read_bio_X509(cert_bio, NULL, NULL, NULL)) != NULL) {
         if (sk_X509_push(chain, cert) <= 0) {
+#endif // OPENSSL_IS_BORINGSSL
+
             tcn_Throw(e, "No Certificate specified or invalid format");
             goto cleanup;
         }
+
+#ifndef OPENSSL_IS_BORINGSSL
         cert = NULL;
+#endif // OPENSSL_IS_BORINGSSL
```

Comment:
What do you think about duplicating a little bit of common logic instead of ending the `ifdef` block with an unfinished `if` and `while` block?

Reply:
I actually would prefer to keep the `#ifdef` stuff to ensure we do not miss to fix something because of code duplication.

Reply:
How does IDE treats you in this case? If IDEs are happy with this, then I think it's OK. But [this one](https://github.com/netty/netty-tcnative/pull/445#discussion_r265394786) still could be addressed since it has no duplicate.

Reply:
CLion is happy, but that said I usually use vi for the C stuff ;)

Reply:
Haha, OK, then. But could you take care of [this one](https://github.com/netty/netty-tcnative/pull/445#discussion_r265394786) at least?

Reply:
@trustin done